openmetadata-ui/src/main/resources/ui/src/components/PermissionProvider/PermissionProvider.tsx

```diff
@@ -0,0 +1,89 @@
+/*
+ *  Copyright 2021 Collate
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+import { AxiosError } from 'axios';
+import { observer } from 'mobx-react';
+import React, {
+  createContext,
+  FC,
+  ReactNode,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+import AppState from '../../AppState';
+import { getLoggedInUserPermissions } from '../../axiosAPIs/miscAPI';
+import { getPermissionMap } from '../../utils/PermissionsUtils';
+import { showErrorToast } from '../../utils/ToastUtils';
+import { PermissionMap } from './PermissionProvider.interface';
+
+/**
+ * Permission Context
+ * Returns ResourcePermission List for loggedIn User
+ * @returns PermissionMap
+ */
+export const PermissionContext = createContext<{
+  permissions: PermissionMap;
+}>({ permissions: {} as PermissionMap });
+
+interface PermissionProviderProps {
+  children: ReactNode;
+}
+
+/**
+ *
+ * @param children:ReactNode
+ * @returns JSX
+ */
+const PermissionProvider: FC<PermissionProviderProps> = ({ children }) => {
+  const [permissions, setPermissions] = useState<PermissionMap>(
+    {} as PermissionMap
+  );
+
+  // Update current user details of AppState change
+  const currentUser = useMemo(() => {
+    return AppState.getCurrentUserDetails();
+  }, [AppState.userDetails, AppState.nonSecureUserDetails]);
+
+  /**
+   * Fetch permission for logged in user
+   */
+  const fetchLoggedInUserPermissions = async () => {
+    try {
+      const response = await getLoggedInUserPermissions();
+      setPermissions(getPermissionMap(response.data || []));
+    } catch (error) {
+      showErrorToast(error as AxiosError);
+    }
```

Comment:
Currently Provider is doing API call and stores parsed data. Let's enhance provider for easy use of Resource level permissions

Reply:
@chirag-madlani , If I understood correctly, here we should pass resource name and fetch permission for the same, is that what you meant?

Reply:
Or it might provide all the Permission we needed. So when new permission comes in we need to modify this provider only

Reply:
@chirag-madlani , Here we are providing all the permission like this format.
```js
{rosourceName: permissions[]}
```